webapp/src/components/SampleTablePrime.vue

```diff
@@ -0,0 +1,275 @@
+</template>
+
+<script>
+import CreateItemModal from "@/components/CreateItemModal";
+import BatchCreateSampleModal from "@/components/BatchCreateSampleModal";
+
+import DataTable from "primevue/datatable";
+import Column from "primevue/column";
+import Button from "primevue/button";
+import IconField from "primevue/iconfield";
+import InputIcon from "primevue/inputicon";
+import InputText from "primevue/inputtext";
+import { FilterMatchMode, FilterOperator } from "@primevue/core/api";
+import "primeicons/primeicons.css";
+
+import { getSampleList, deleteSample } from "@/server_fetch_utils.js";
+import FormattedItemName from "@/components/FormattedItemName";
+import ChemicalFormula from "@/components/ChemicalFormula";
+import CollectionList from "@/components/CollectionList";
+import Creators from "@/components/Creators";
+
+export default {
+  components: {
+    CreateItemModal,
+    BatchCreateSampleModal,
+    DataTable,
+    Column,
+    Button,
+    IconField,
+    InputIcon,
+    InputText,
+    FormattedItemName,
+    ChemicalFormula,
+    CollectionList,
+    Creators,
+  },
+  data() {
+    return {
+      createItemModalIsOpen: false,
+      batchCreateSampleModalIsOpen: false,
+      isSampleFetchError: false,
+      itemsSelected: [],
+      expandedRows: [],
+      availableColumns: [
+        { field: "item_id", header: "ID" },
+        { field: "type", header: "Type" },
+        { field: "name", header: "Sample name" },
+        { field: "chemform", header: "Formula" },
+        { field: "date", header: "Date" },
+        {
+          field: "collections",
+          header: "Collections",
+        },
+        { field: "creators", header: "Creators" },
+        { field: "nblocks", header: "# of blocks" },
+      ],
+      filters: {
+        global: { value: null },
+        item_id: {
+          operator: FilterOperator.AND,
+          constraints: [{ value: null, matchMode: FilterMatchMode.CONTAINS }],
+        },
+        type: {
+          operator: null,
+          constraints: [{ value: null, matchMode: FilterMatchMode.CONTAINS }],
+        },
+      },
+    };
+  },
+  computed: {
+    samplesWithComputedFields() {
+      return this.$store.state.sample_list.map((sample) => ({
+        ...sample,
+        collectionsList: sample.collections.map((collection) => collection.collection_id).join(" "),
+        creatorsList: sample.creators.map((creator) => creator.display_name).join(" "),
+      }));
+    },
+    samples() {
+      return this.samplesWithComputedFields;
+    },
+  },
+  methods: {
+    onFilter: function (event) {
+      console.log(event.filteredValue);
+    },
+    goToEditPage(event) {
+      const row = event.data;
+      if (
+        event.originalEvent.ctrlKey ||
+        event.originalEvent.metaKey ||
+        event.originalEvent.altKey
+      ) {
+        window.open(`/edit/${row.item_id}`, "_blank");
+      } else {
+        this.$router.push(`/edit/${row.item_id}`);
+      }
+    },
```

Comment:
Empirically there is something strange going on here, when click on an ID (specifically) in the table, it opens in a new tab, and also tries to open it in the old tab (but fails with some JS error)